internal/provider/provider.go

```diff
@@ -0,0 +1,253 @@
+package provider
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+	vaultclient "github.com/hashicorp/secrets-store-csi-driver-provider-vault/internal/client"
+	"github.com/hashicorp/secrets-store-csi-driver-provider-vault/internal/config"
+	"github.com/hashicorp/vault/api"
+	"github.com/mitchellh/mapstructure"
+)
+
+func readJWTToken(path string) (string, error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", fmt.Errorf("failed to read jwt token: %w", err)
+	}
+
+	return string(bytes.TrimSpace(data)), nil
+}
+
+// provider implements the secrets-store-csi-driver provider interface
+// and communicates with the Vault API.
+type provider struct {
+	logger hclog.Logger
+}
+
+func NewProvider(logger hclog.Logger) *provider {
+	p := &provider{
+		logger: logger,
+	}
+
+	return p
+}
+
+func (p *provider) getMountInfo(ctx context.Context, client *api.Client, mountName string) (string, string, error) {
```

Comment:
Do we need this for anything else besides checking if a secret mount is KV v2? If not we can tell if it's KV v2 by looking at the payload. If it contains `.data.data`, then it's KV v2. If we did this instead that's one less permission CSI will need.

Reply:
Yeah that's correct. I'll strip this out when adding generic secret support.